File: sdk/javascript/paissive_income_sdk/services/ai-models.js
/**
 * AI Models service for the pAIssive Income API.
 * 
 * This module provides a service for interacting with the AI model endpoints.
 */

const BaseService = require('./base');

/**
 * AI Models service.
 * @extends BaseService
 */
class AIModelsService extends BaseService {
  /**
   * Get all available AI models.
   * 
   * @returns {Promise<Object>} List of AI models
   */
  async getModels() {
    return this._get('ai-models/models');
  }
  
  /**
   * Get details for a specific AI model.
   * 
   * @param {string} modelId - Model ID
   * @returns {Promise<Object>} Model details
   */
  async getModel(modelId) {
    return this._get(`ai-models/models/${modelId}`);
  }
  
  /**
   * Run inference on an AI model.
   * 
   * @param {Object} data - Inference request data
   * @param {string} data.model_id - Model ID
   * @param {Object} data.inputs - Input data for the model
   * @param {Object} data.parameters - Inference parameters
   * @returns {Promise<Object>} Inference results
   */
  async runInference(data) {
    return this._post('ai-models/inference', data);
  }
  
  /**
   * Download or register an AI model.
   * 
   * @param {Object} data - Model download data
   * @param {string} data.name - Model name
   * @param {string} data.source - Model source
   * @param {string} data.version - Model version
   * @returns {Promise<Object>} Download status
   */
  async downloadModel(data) {
    return this._post('ai-models/models/download', data);
  }
  
  /**
   * Get performance metrics for a specific AI model.
   * 
   * @param {string} modelId - Model ID
   * @returns {Promise<Object>} Performance metrics
   */
  async getModelPerformance(modelId) {
    return this._get(`ai-models/models/${modelId}/performance`);
  }
  
  /**
   * Compare multiple AI models.
   * 
   * @param {Object} data - Model comparison data
   * @param {Array<string>} data.model_ids - List of model IDs to compare
   * @param {Array<string>} data.metrics - List of metrics to compare
   * @param {Object} data.test_data - Optional test data for evaluation
   * @returns {Promise<Object>} Comparison results
   */
  async compareModels(data) {
    return this._post('ai-models/models/compare', data);
  }
  
  /**
   * Create a new version of an AI model.
   * 
   * @param {string} modelId - Model ID
   * @param {Object} data - Version creation data
   * @param {string} data.version - Version name
   * @param {string} data.changes - Description of changes
   * @param {Object} data.parameters - Model parameters
   * @returns {Promise<Object>} Created model version
   */
  async createModelVersion(modelId, data) {
    return this._post(`ai-models/models/${modelId}/versions`, data);
  }
}

module.exports = AIModelsService;

File: sdk/python/paissive_income_sdk/services/ai_models.py
"""
"""
AI Models service for the pAIssive Income API.
AI Models service for the pAIssive Income API.


This module provides a service for interacting with the AI model endpoints.
This module provides a service for interacting with the AI model endpoints.
"""
"""




from typing import Any, Dict
from typing import Any, Dict


from .base import BaseService
from .base import BaseService




class AIModelsService:
    class AIModelsService:


    pass  # Added missing block
    pass  # Added missing block
    """
    """
    AI Models service.
    AI Models service.
    """
    """


    def get_models(self) -> Dict[str, Any]:
    def get_models(self) -> Dict[str, Any]:
    """
    """
    Get all available AI models.
    Get all available AI models.


    Returns:
    Returns:
    List of AI models
    List of AI models
    """
    """
    return self._get("ai-models/models")
    return self._get("ai-models/models")


    def get_model(self, model_id: str) -> Dict[str, Any]:
    def get_model(self, model_id: str) -> Dict[str, Any]:
    """
    """
    Get details for a specific AI model.
    Get details for a specific AI model.


    Args:
    Args:
    model_id: Model ID
    model_id: Model ID


    Returns:
    Returns:
    Model details
    Model details
    """
    """
    return self._get(f"ai-models/models/{model_id}")
    return self._get(f"ai-models/models/{model_id}")


    def run_inference(self, data: Dict[str, Any]) -> Dict[str, Any]:
    def run_inference(self, data: Dict[str, Any]) -> Dict[str, Any]:
    """
    """
    Run inference on an AI model.
    Run inference on an AI model.


    Args:
    Args:
    data: Inference request data
    data: Inference request data
    - model_id: Model ID
    - model_id: Model ID
    - inputs: Input data for the model
    - inputs: Input data for the model
    - parameters: Inference parameters
    - parameters: Inference parameters


    Returns:
    Returns:
    Inference results
    Inference results
    """
    """
    return self._post("ai-models/inference", data)
    return self._post("ai-models/inference", data)


    def download_model(self, data: Dict[str, Any]) -> Dict[str, Any]:
    def download_model(self, data: Dict[str, Any]) -> Dict[str, Any]:
    """
    """
    Download or register an AI model.
    Download or register an AI model.


    Args:
    Args:
    data: Model download data
    data: Model download data
    - name: Model name
    - name: Model name
    - source: Model source
    - source: Model source
    - version: Model version
    - version: Model version


    Returns:
    Returns:
    Download status
    Download status
    """
    """
    return self._post("ai-models/models/download", data)
    return self._post("ai-models/models/download", data)


    def get_model_performance(self, model_id: str) -> Dict[str, Any]:
    def get_model_performance(self, model_id: str) -> Dict[str, Any]:
    """
    """
    Get performance metrics for a specific AI model.
    Get performance metrics for a specific AI model.


    Args:
    Args:
    model_id: Model ID
    model_id: Model ID


    Returns:
    Returns:
    Performance metrics
    Performance metrics
    """
    """
    return self._get(f"ai-models/models/{model_id}/performance")
    return self._get(f"ai-models/models/{model_id}/performance")


    def compare_models(self, data: Dict[str, Any]) -> Dict[str, Any]:
    def compare_models(self, data: Dict[str, Any]) -> Dict[str, Any]:
    """
    """
    Compare multiple AI models.
    Compare multiple AI models.


    Args:
    Args:
    data: Model comparison data
    data: Model comparison data
    - model_ids: List of model IDs to compare
    - model_ids: List of model IDs to compare
    - metrics: List of metrics to compare
    - metrics: List of metrics to compare
    - test_data: Optional test data for evaluation
    - test_data: Optional test data for evaluation


    Returns:
    Returns:
    Comparison results
    Comparison results
    """
    """
    return self._post("ai-models/models/compare", data)
    return self._post("ai-models/models/compare", data)


    def create_model_version(
    def create_model_version(
    self, model_id: str, data: Dict[str, Any]
    self, model_id: str, data: Dict[str, Any]
    ) -> Dict[str, Any]:
    ) -> Dict[str, Any]:
    """
    """
    Create a new version of an AI model.
    Create a new version of an AI model.


    Args:
    Args:
    model_id: Model ID
    model_id: Model ID
    data: Version creation data
    data: Version creation data
    - version: Version name
    - version: Version name
    - changes: Description of changes
    - changes: Description of changes
    - parameters: Model parameters
    - parameters: Model parameters


    Returns:
    Returns:
    Created model version
    Created model version
    """
    """
    return self._post(f"ai-models/models/{model_id}/versions", data)
    return self._post(f"ai-models/models/{model_id}/versions", data)